### PR TITLE
Chapter 11 implementation

### PR DESCRIPTION
**Pull request by czegreanu**
ViewPager integration

### CODE REVIEW

File: app/src/main/java/zegreanu/cristi/criminalintent/CrimePagerActivity.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class CrimePagerActivity : AppCompatActivity() {
+    private lateinit var viewPager: ViewPager
+    private var crimes = arrayListOf<Crime>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_crime_pager)
+
+        val a = findViewById<ViewPager>(R.id.crime_view_pager)
```
Comment (deyupopp) [inline app/src/main/java/zegreanu/cristi/criminalintent/CrimePagerActivity.kt]: why do we need this variable?

File: app/src/main/java/zegreanu/cristi/criminalintent/CrimePagerActivity.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        viewPager = a
+        val crimeId = intent.getSerializableExtra(EXTRA_CRIME_ID)
+        crimes = CrimeLab.crimes
+
+        viewPager.adapter = CrimeSlidePagerAdapter(crimes, supportFragmentManager)
+        viewPager.currentItem = crimes.indexOf(CrimeLab[crimeId as UUID])
```
Comment (deyupopp) [inline app/src/main/java/zegreanu/cristi/criminalintent/CrimePagerActivity.kt]: Saw this before, and maybe it's worth having an extension on intent, called getUUID(key: String), and do the cast inside

File: app/src/main/java/zegreanu/cristi/criminalintent/CrimePagerActivity.kt (hunk 1 above)
Comment (czegreanu) [reply app/src/main/java/zegreanu/cristi/criminalintent/CrimePagerActivity.kt]: Left it by mistake, used for debugging

File: app/src/main/res/layout/activity_crime_pager.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.viewpager.widget.ViewPager xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (cosminstirbu) [inline app/src/main/res/layout/activity_crime_pager.xml]: Hmm, maybe we can use https://developer.android.com/jetpack/androidx/releases/viewpager2

**Review by cosminstirbu — CHANGES_REQUESTED** [review]

**Review by cosminstirbu — APPROVED** [review]

**Pull request merged**